src/resolvers/userResolver.js

```diff
@@ -134,6 +134,62 @@ const userResolver = {
       );
     },
 
+    changeUsername: async(_, { userInfo }, { req }) => {
+      const { newUsername } = userInfo;
+      const oldUser = await User.findById(req.session.userId);
+      const oldUsername = oldUser.username;
+
+      if(oldUsername === newUsername) {
+        return {
+          errors: [
+            {
+              field: "username",
+              message: "old and new username are same",
+            },
+          ],
+        };
+      }
+
+      let user = await User.findOne({ username: oldUsername });
```

Comment:
The `user` is fetched already at line 139 so we don't need to do that again. We already have to `userId` of the user so we don't need to check whether the old username exists. We can use the `userId` to update the user.